cmd/status.go

```diff
@@ -0,0 +1,84 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/exoscale/cli/table"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	statusURL         = "https://exoscalestatus.com/api.json"
+	statusContentPage = "application/json"
+)
+
+// statusCmd represents the status command
+var statusCmd = &cobra.Command{
+	Use:   "status",
+	Short: "Exoscale status",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		r, err := http.Get(statusURL)
+		if err != nil {
+			return err
+		}
+		defer r.Body.Close()
+
+		contentType := r.Header.Get("content-type")
+		if contentType != statusContentPage {
+			return fmt.Errorf("status page content type expected %q, but got %q", statusContentPage, contentType)
+		}
+
+		response := &ExoscaleStatus{}
+		if err := json.NewDecoder(r.Body).Decode(response); err != nil {
+			return err
+		}
+
+		tableWriter := table.NewTable(os.Stdout)
+
+		tableWriter.SetHeader([]string{"Detailed Status", "State"})
+
+		tableWriter.Append([]string{"Compute", response.Status.Compute.State})
+		tableWriter.Append([]string{"Compute API", response.Status.ComputeAPI.State})
+		tableWriter.Append([]string{"DNS", response.Status.DNS.State})
+		tableWriter.Append([]string{"Object Storage", response.Status.ObjectStorage.State})
+
+		tableWriter.Render()
+
+		tableWriter = table.NewTable(os.Stdout)
+
+		tableWriter.SetHeader([]string{"Current Events", "Upcoming Maintenances"})
+
+		tableWriter.Render()
+
+		return nil
+	},
+}
+
+//ExoscaleStatus represent exoscale statsus
+type ExoscaleStatus struct {
+	URL                  string        `json:"url"`
+	Incidents            []interface{} `json:"incidents"`
+	UpcomingMaintenances []interface{} `json:"upcoming_maintenances"`
```

Comment:
In `Json` body I have not example of `incidents ` or `upcoming_maintenances` so I don't know how to get this infos (string, other...?) because it's empty, e.g: `"incidents": []`

Reply:
probably there, https://community.exoscale.com/api/runstatus/ or ask @brutasse

Reply:
Some inspiration in our fingerd daemon: https://github.com/exoscale/runstatus-fingerd/blob/master/src/finger/engine.clj